src/script/standard.cpp

```diff
@@ -239,49 +239,43 @@ bool ExtractDestinations(const CScript& scriptPubKey, txnouttype& typeRet, std::
 
 namespace
 {
-class CScriptVisitor : public boost::static_visitor<bool>
+class CScriptVisitor : public boost::static_visitor<void>
 {
 private:
-    CScript *script;
+    CScript& script;
 public:
-    explicit CScriptVisitor(CScript *scriptin) { script = scriptin; }
+    explicit CScriptVisitor(CScript& scriptin) : script(scriptin) {}
 
-    bool operator()(const CNoDestination &dest) const {
-        script->clear();
-        return false;
```

Comment:
I'm curious why `return false` here.

Reply:
Sometimes code is written to have a return type even though the return type is initially unused. Motivation for that could be that the initial implementation wants to accommodate future callers of the code, which may want to use the return type.

Reply:
Yeah, I think that motivation is long gone.